yt_dlp/extractor/huya.py

```diff
@@ -55,10 +55,8 @@ class HuyaLiveIE(InfoExtractor):
     def _real_extract(self, url):
         video_id = self._match_id(url)
         webpage = self._download_webpage(url, video_id=video_id)
-        json_stream = self._search_regex(r'"stream":\s+"([a-zA-Z0-9+=/]+)"', webpage, 'stream', default=None)
-        if not json_stream:
-            raise ExtractorError('Video is offline', expected=True)
-        stream_data = self._parse_json(compat_b64decode(json_stream).decode(), video_id=video_id,
+        json_stream = self._search_regex(r'stream:\s+({.*})', webpage, 'stream', default=None)
+        stream_data = self._parse_json(json_stream, video_id=video_id,
```

Comment:
Use _search_json

Reply:
Done. Had to squash some changes due to an unnecessary import, but this should work now.